src/lightning/fabric/wrappers.py

```diff
@@ -263,7 +263,7 @@ def __setattr__(self, name: str, value: Any) -> None:
         original_has_attr = hasattr(original_module, name)
         # Can't use super().__getattr__ because nn.Module only checks _parameters, _buffers, and _modules
         # Can't use self.__getattr__ because it would pass through to the original module
-        fabric_has_attr = name in self.__dict__
+        fabric_has_attr = hasattr(self, name)
```

Comment:
If the comment above is true, then we won't be able to do this. Can you check?

Reply:
I suggest to do the monkey patching on the original model, by accessing the wrapped module first.

Reply:
Sorry, what should I check precisely?

I actually want to monkeypatch the `_FabricModule`: https://github.com/Lightning-AI/lit-thunder-LEGACY/issues/2397

Reply:
I'm referring to the comment above the line:

> Can't use self.__getattr__ because it would pass through to the original module

which explains why we're not calling hasattr, and access self.dict instead.

Reply:
Maybe checking `name in dir(self)` could help here?

Reply:
Pushed an alternative implementation and extended an existing test case to ensure that `hasattr()` would've failed